debezium/src/main/scala/compstak/kafkastreams4s/debezium/DebeziumTable.scala

```diff
@@ -6,13 +6,27 @@ import org.apache.kafka.common.serialization.Serde
 import compstak.kafkastreams4s.circe.CirceSerdes
 import compstak.circe.debezium.{DebeziumKey, DebeziumValue}
 import io.circe.{Decoder, Encoder}
+import cats.FunctorFilter
+import cats.Functor
 
 case class DebeziumTable[K, V](
   topicName: String,
   idName: String,
   toKTable: KTable[DebeziumKey[K], V]
 ) {
 
+  def map[V2](f: V => V2): DebeziumTable[K, V2] =
+    copy(toKTable = toKTable.mapValues(v => f(v)))
+
+  def mapWithKey[V2](f: (DebeziumKey[K], V) => V2): DebeziumTable[K, V2] =
+    copy(toKTable = toKTable.mapValues((k, v) => f(k, v)))
+
+  def mapFilter[V2](f: V => Option[V2]): DebeziumTable[K, V2] =
+    copy(toKTable = toKTable.mapValues(v => f(v)).filter((k, o) => o.isDefined).mapValues(_.get))
+
+  def mapFilterWithKey[V2](f: (DebeziumKey[K], V) => Option[V2]): DebeziumTable[K, V2] =
+    copy(toKTable = toKTable.mapValues((k, v) => f(k, v)).filter((k, o) => o.isDefined).mapValues(_.get))
```

Comment:
in these `mapValues`, can't we just do `mapValue(f)`?

Reply:
Nope, it's a Java API and it uses a custom class, that scalac can't coerce without writing it as a lambda :)